Bluetooth: Conditionally include logging Kconfigs

jori-nordic opened the pull request:
```
Bluetooth: Re-organize logging Kconfig options

- got rid of "Others" menu
- JK I made a new one: "Common"
- moved all host-related opts to "Host" instead of "Others"
- moved all audio-related opts behind `if BT_AUDIO`
- moved only controller kconfig to controller/Kconfig.ll_sw_split
  - it had a dependency there already
```
```
Bluetooth: Conditionally include logging Kconfigs

Having all the configs be visible by default clutters the menuconfig
view. Make them visible only when the corresponding subsys or module is
enabled.
```

Fixes https://github.com/zephyrproject-rtos/zephyr/issues/76863

[log-options.webm](https://github.com/user-attachments/assets/97b94eb2-254b-4ca7-9096-3169d2ab31da)

jori-nordic committed:
Bluetooth: Conditionally include logging Kconfigs

Having all the configs be visible by default clutters the menuconfig
view. Make them visible only when the corresponding subsys or module is
enabled.

Signed-off-by: Jonathan Rico <jonathan.rico@nordicsemi.no>

jori-nordic committed:
Bluetooth: Re-organize logging Kconfig options

- got rid of "Others" menu
- JK I made a new one: "Common"
- moved all host-related opts to "Host" instead of "Others"
- moved all audio-related opts behind `if BT_AUDIO`
- moved only controller kconfig to controller/Kconfig.ll_sw_split
  - it had a dependency there already

Signed-off-by: Jonathan Rico <jonathan.rico@nordicsemi.no>

Thalley commented on subsys/bluetooth/Kconfig.logging:
```diff
@@ -28,267 +28,344 @@ parent-module = BT
+source "subsys/logging/Kconfig.template.log_config_inherit"
+endif # BT_SMP
+
+module = BT_SERVICE
+module-str = "Bluetooth Services"
+source "subsys/logging/Kconfig.template.log_config_inherit"
+endmenu # LE Host
+endif # BT_HCI_HOST
```
This is only used for DIS, right? Could guard it with it since e.g. OTS and BAS have their own

jori-nordic replied:
Could be used for other non-audio services later? idk

Thalley commented on subsys/bluetooth/Kconfig.logging:
```diff
@@ -28,267 +28,344 @@ parent-module = BT
 source "subsys/logging/Kconfig.template.log_config_inherit"
+endif # BT_CSIP_SET_COORDINATOR
 
+if BT_AUDIO
 module = BT_CSIP_SET_MEMBER_CRYPTO
 module-str = "Coordinated Set Identification Profile crypto functions"
 source "subsys/logging/Kconfig.template.log_config_inherit"
+endif # BT_AUDIO
```
Can we do an || / "OR" here? 

Should only be available if `BT_CSIP_SET_MEMBER` or `BT_CSIP_SET_COORDINATOR`

jori-nordic replied:
Could you make a hidden kconfig for this then?

jori-nordic committed:
Bluetooth: OTS: Use proper kconfig log option

`OTS_CLIENT=y` and `OTS=n` is a valid config.

Since the logs weren't conditionally included, this flew under the radar
for a while.

Signed-off-by: Jonathan Rico <jonathan.rico@nordicsemi.no>

jori-nordic committed:
Bluetooth: audio: Make unit tests compile

The tests manually include sources that are normally conditionally
included (ie based on kconfig values) in the stack.

The tests should be fixed by the original authors.
For now, force-on the hidden kconfig that enables the log options.

Signed-off-by: Jonathan Rico <jonathan.rico@nordicsemi.no>

Thalley commented on tests/bluetooth/audio/bap_base/Kconfig:
```diff
@@ -0,0 +1,14 @@
+# Temporary override
+#
+# Copyright (c) 2024 Nordic Semiconductor ASA
+
+# SPDX-License-Identifier: Apache-2.0
+#
+
+config BT_BAP_BASE
+	# Override until someone fixes the UT build system
+	bool
+	default y
+
+# Include Zephyr's Kconfig.
+source "Kconfig.zephyr"
```
Welp :(

Will have a look at whatever caused these things once this is merged

Thalley replied:
https://github.com/zephyrproject-rtos/zephyr/pull/77406
https://github.com/zephyrproject-rtos/zephyr/pull/77405
